[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/drbob-richardson/stat220/blob/main/Assignments/Stat_220_HW2.ipynb)

**Problem 1** A baseball player has a batting average of .240 against right handed pitchers and .375 against left handed pitchers. A batting average can be thought of as the probability they will get a hit.

Part a. What is the probability the baseball player gets 3 out of 4 hits against a right handed pitcher?

In [1]:
import math

batting_average_right_handed = 0.24

total_at_bats = 4

hits_wanted = 3

def calculate_probability(n, k, p):
    binomial_coefficient = math.comb(n, k)
    probability = binomial_coefficient * (p ** k) * ((1 - p) ** (n - k))
    return probability

probability = calculate_probability(total_at_bats, hits_wanted, batting_average_right_handed)

print(f"The probability of getting 3 out of 4 hits against a right-handed pitcher is: {probability}")

The probability of getting 3 out of 4 hits against a right-handed pitcher is: 0.04202495999999999


Part b. What is the probability the baseball gets 3 out of 4 hits against a left handed pitcher.

In [2]:
batting_average_left_handed = 0.375

total_at_bats = 4

hits_wanted = 3
probability_l= calculate_probability(total_at_bats, hits_wanted, batting_average_left_handed)
print(f"The probability of getting 3 out of 4 hits against a left-handed pitcher is: {probability_l}")

The probability of getting 3 out of 4 hits against a left-handed pitcher is: 0.1318359375


Part c. The baseball player gets 3 out of 4 hits against a randomly chosen pitcher. Based on these results, what would you assume about what hand the pitcher uses?

Since the sample size is only 4 hits. It's important to consider that a small number of at-bats may not provide a complete picture of a pitcher's true ability. A larger sample size of at-bats would be more accurative.

Part d. Assume that the pitcher is randomly chosen from the team where 80% of the pitchers are right handed. Use Bayes rule to determine the probability the pitcher is right handed given that the batter hits 3 out of 4 pitches.

In [3]:
prior_probability_right_handed = 0.8
prior_probability_left_handed = 1 - prior_probability_right_handed

likelihood_right_handed = (0.24 ** 3) * (0.76 ** 1)
likelihood_left_handed = (0.375 ** 3) * (0.625 ** 1)

evidence = (prior_probability_right_handed * likelihood_right_handed) + (prior_probability_left_handed * likelihood_left_handed)

posterior_probability_right_handed = (prior_probability_right_handed * likelihood_right_handed) / evidence

print(posterior_probability_right_handed)

0.5604527789286492


**Problem 2** The probability of rain in Arizona on a randomly chosen day of the year is 5%. A researcher started a study on rain in Arizona and noted 3 days in a row of rain. The researcher calculated the probability of this in the following way:

Pr(3 days of rain) = Pr(1 day of rain)^3 = .05^3 = .000125

The researcher was so excited about finding such a rare event that he published his results. You write a scathing review of his publication stating he made a poor assumption about rain. What did the researcher get wrong?


The probability of three consecutive rainy days is not equal to the probability of a single rainy day raised to the third power. His assumption is incorrect because it overlooks the fact that weather events are not independent. Calculate the probability of multiple events occurring together by multiplying the probabilities of each individual event only applys on independent event.

**Problem 3** A stock trader is examining stocks. Let event A be the event that Stock A falls. Let event B be the event that Stock B falls. Assume the following.

*   Pr(A) = 0.4
*   Pr(B) = 0.25
*   Pr(A and B) = 0.15


Part a. What is the probability that either events A or B happen?

In [4]:
P_A = 0.4
P_B = 0.25
P_A_given_B = 0.5


P_A_and_B = P_A_given_B * P_B


print(f"The probability that both events A and B both happen is: {P_A_and_B}")

The probability that both events A and B both happen is: 0.125


Part b. From this, can you make statement as to whether or not events A and B are mutually exclusive?

Two events are mutually exclusive if they cannot happen at the same time, meaning  P(A and B) = 0 .

In this case,  P(A and B) = 0.125 , which means both events can happen simultaneously. Therefore, events A and B are not mutually exclusive.

Part c. What can you say about if events A and B are independent?

If they are independent, P A and B = P A * P B
PA_and_B = 0.125
PA * PB = 0.1


**Problem 4** Each time a person shops at a grocery store, the event of catching a cold or some other virus from another shopper is independent from visit to visit and has a constant probability over the year, equal to 0.01.

Part a. In 100 trips to this store over the course of a year, the probability of catching a virus while shopping there is 100(0.01) = 1.0. What is wrong with this reasoning?

it assumes that the probability of catching a virus on any one trip is simply additive across multiple trips, which is not how probabilities work for independent events.

Part b. Find the correct probability in (a).

In [ ]:
probability_single_trip = 0.01

probability_not_single_trip = 1 - probability_single_trip

probability_not_100_trips = (probability_not_single_trip) ** 100

probability_at_least_once = 1 - probability_not_100_trips

print(f"The correct probability of catching a virus at least once in 100 trips is: {probability_at_least_once}")


Part c. Determine the expected number of days someone would go to a grocery store before they catch a cold or some other virus.

In [ ]:
p_virus = 0.01

expected_days = 1 / p_virus

print(f"The expected number of days before catching a virus is: {expected_days}")

**Problem 5** A quiz has twelve multiple-choice questions, with four possible answers for each. A student who is completely unprepared randomly guesses the answer for each question. Let Y denote the number of correct responses

Part a. What is the name of the distribution that can be used to determine the number of questionsd guessed correctly? What are the parameters of this distribution?

Name of the distribution: Binomial distribution
Parameters:
n=12 total number of questions
p=1/4 the probability of guessing a question correctly since there are 4 possible answers

Part b. Find the probability that the students gets (i) all 12, (ii) none of the questions correct.

In [ ]:
from math import comb

n = 12

p = 1 / 4

probability_all_correct = (p ** n)

probability_none_correct = ((1 - p) ** n)

# Print the results
print(f"(i) Probability of getting all 12 questions correct: {probability_all_correct:.10f}")
print(f"(ii) Probability of getting none of the questions correct: {probability_none_correct:.10f}")


Part c. What is the expected value and variance of the number of questions guessed correctly?

In [ ]:
mean_Y = n * p

std_dev_Y = (n * p * (1 - p)) ** 0.5

print(f"Mean (expected value) of Y: {mean_Y:.2f}")
print(f"Standard deviation of Y: {std_dev_Y:.2f}")